src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -205,15 +190,21 @@ public static Appointment parseSerialisedAppointment(String appointment) throws
     }
 
     /**
-     * Parses {@code Collection<String> allergies} into a {@code Set<Allergy>}.
+     * Parses and validates the given allergy string.
+     *
+     * @param allergyStr The medical condition string to parse.
+     * @return A {@code Allergy} object.
+     * @throws ParseException If the allergy string is invalid.
      */
-    public static Set<Allergy> parseAllergies(Collection<String> allergies) throws ParseException {
-        requireNonNull(allergies);
-        final Set<Allergy> allergiesSet = new HashSet<>();
-        for (String allergyName : allergies) {
-            allergiesSet.add(parseAllergy(allergyName));
```

Comment:
This was how the original `parseTags` was done, I am not sure how your new `parseAllergy` will affect the command